### ISSUE TITLE
LINQ Contains method on Non Key column in Where clause is not translated to query

### ISSUE THREAD

**Issue by jyothi530** [dotnet/efcore#18627]
Writing a LINQ query with Contains function on a non key column throwing InvalidOperationException . Cannot translate to query.

### Steps to reproduce

Write any simple query on a DbSet Model with Where clause filtering on a Non-Key column with Contains functions.

```
System.InvalidOperationException: The LINQ expression 'Where<Activity>(
          source: DbSet<Activity>,
          predicate: (a) => (Unhandled parameter: __delCenterStateId_0).Contains(a.DeliveryCenterActivityStateId))' could not be translated. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to either AsEnumerable(), AsAsyncEnumerable(), ToList(), or ToListAsync(). See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.
         at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.<VisitMethodCall>g__CheckTranslated|8_0(ShapedQueryExpression translated, <>c__DisplayClass8_0& )
         at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
         at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
         at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
         at Microsoft.EntityFrameworkCore.Query.QueryCompilationContext.CreateQueryExecutor[TResult](Expression query)
         at Microsoft.EntityFrameworkCore.Storage.Database.CompileQuery[TResult](Expression query, Boolean async)
         at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](IDatabase database, Expression query, IModel model, Boolean async)
         at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass9_0`1.<Execute>b__0()
         at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
         at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)
         at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
         at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)
         at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1.GetEnumerator()
         at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
         at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
```

### Further technical details

EF Core version: 3.0
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.0
Operating system: Windows 10
IDE: Visual Studio 2019 16.3

**Comment by ajcvickers** [comment]
@jyothi530 This looks like a duplicate of #17342. If it isn't, then please post a small, runnable project/solution or complete code listing that demonstrates the behavior you are seeing.

**Comment by jyothi530** [comment]
@ajcvickers Yes. I think it is a duplicate of #17342. Looks like stack trace is different. You can close this one if it is a duplicate. I believe this should be fixed as part of 3.1 right. Thanks

**Comment by ajcvickers** [comment]
Correct; fixed for 3.1
